spipe/main.c

```diff
@@ -222,20 +229,28 @@ main(int argc, char * argv[])
 	ET.connection_error = 0;
 	ET.stopped = 0;
 
+	/* Resolve bind address. */
+	if (opt_b) {
+	    if ((sa_b = sock_resolve(opt_b)[0]) == NULL) {
```

Comment:
Need to store the complete list returned by `sock_resolve` in a variable.

Check that it's not NULL (in case `sock_resolve` failed).

And free the entire list later (not just the one address we end up using).

Reply:
Done, for spiped too.